libbeat/tests/integration/framework.go

```diff
@@ -123,6 +168,8 @@ func (b *BeatProc) Start(args ...string) {
 			// 4. Wait for the goroutine to finish, this helps ensuring
 			// no other Beat process was started
 			wg.Wait()
+		} else {
+			b.cmdMutex.Unlock()
```

Comment:
maybe. @belimawr has an Unlock() happening in the `if` part above.  I'd like them to comment to see if there was a reason not to just `defer` the Unlock().

Reply:
Yes, there is a good reason, we need to release the mutex before calling `wg.Wait()` because the goroutine https://github.com/elastic/beats/blob/63e88307aa982712979822ca0262beaefef0de61/libbeat/tests/integration/framework.go#L140-L146 might call `b.startBeat()` and this will lock the mutex, so if it's not released before calling `wg.Wait()` we can end up on a deadlock.

We could improve the comment https://github.com/elastic/beats/blob/63e88307aa982712979822ca0262beaefef0de61/libbeat/tests/integration/framework.go#L165-L166 explaining this deadlock scenario.

Reply:
@AndersonQ I updated the comment to reflect info that @belimawr gave.  Can you give it another look?